setup.py

```diff
@@ -0,0 +1,127 @@
+import os
+import subprocess
+from enum import Enum
+
+from setuptools import setup
+from setuptools.command.test import test
+from Cython.Distutils.extension import Extension
+from Cython.Distutils.build_ext import new_build_ext as build_ext
+from Cython.Build import cythonize
+from distutils.command.install import install
+from distutils.command.build import build
+from packaging import version
+import pip
+import numpy
+
+assert(version.parse(pip.__version__) >= version.parse("10.0.1"))  # minimum version 10.0.1 is required. See https://github.com/precice/precice/wiki/Non%E2%80%93standard-APIs#python-bindings-version-of-pip3-is-too-old
+
+# name of Interfacing API
+APPNAME = "pyprecice"
+# this version should be in sync with the latest supported preCICE version
+precice_version = version.Version("2.0.0")  # todo: should be replaced with precice.get_version(), if possible or we should add an assertion that makes sure that the version of preCICE is actually supported
+# this version number may be increased, if changes for the bindings are required
+bindings_version = version.Version("1")
+APPVERSION = version.Version(str(precice_version) + "." + str(bindings_version))
+
+PYTHON_BINDINGS_PATH = os.path.dirname(os.path.abspath(__file__))
+
+def get_extensions(is_test):
+    compile_args = []
+    link_args = []
+    compile_args.append("-Wall")
```

Comment:
Is the warning flag necessary? If the local `CXX_FLAGS` are used for compiling, then we can simply drop this.

Reply:
I removed this line and `-Wall` still shows up in the compilation command:

```
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.6m -c /home/benjamin/python-bindings/test/test_bindings_module.cpp -o build/temp.linux-x86_64-3.6/home/benjamin/python-bindings/test/test_bindings_module.o -std=c++11 -I/home/benjamin/.local/lib/python3.6/site-packages/numpy/core/include
```